bluej/src/main/java/bluej/editor/base/TextLine.java

```diff
@@ -232,18 +232,21 @@ public void setText(List<StyledSegment> text, double xTranslate, boolean fontCha
         
         hideSelection();
         hideErrorUnderline();
-        getChildren().clear();
-        getChildren().addAll(backgroundNodes);
-        getChildren().addAll(bracketMatchShape, findResultShape, selectionShape);
+        // Since getChildren() has listeners, it's a bit more optimal to construct the list and set it once
+        // rather than manipulate getChildren as we go along:
+        ArrayList<Node> children = new ArrayList<>(5 + text.size());
```

Comment:
That's because I messed it up.  It's for the other bits that get added, but backgroundNodes is itself a list, and I counted it as 1 item.  I've fixed it now.